src/network/syncing_state.cpp

```diff
@@ -0,0 +1,62 @@
+#include "syncing_state.hpp"
+
+namespace taraxa {
+
+void SyncingState::set_peer(const dev::p2p::NodeID& peer_id) {
+  std::unique_lock lock(peer_mutex_);
+  peer_id_ = peer_id;
+}
+
+dev::p2p::NodeID SyncingState::syncing_peer() const {
+  std::shared_lock lock(peer_mutex_);
+  return peer_id_;
+}
+
+void SyncingState::set_pbft_syncing(bool syncing, const std::optional<dev::p2p::NodeID>& peer_id) {
+  assert((syncing && peer_id) || !syncing);
+  pbft_syncing_ = syncing;
+
+  if (peer_id) {
+    set_peer(peer_id.value());
+  }
+  // Reset last sync packet time when syncing is restarted/fresh syncing flag is set
+  set_last_sync_packet_time();
```

Comment:
this does not seem ok ?? I guess it should be called only at the end of pbft packet processing

Reply:
it should be there but only if we are setting syncing to true, now it should be fixed